app/models/user.rb

```diff
@@ -12,4 +13,18 @@ class User < ActiveRecord::Base
 
   has_many :reports,
            dependent: :destroy
+
+  def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
+    data = access_token.info
+    user = User.where(:email => data["email"]).first
+
+    unless user
+      short_name = data["name"].slice!(0..8)
+      user = User.create(username: "user#{User.all.last.id + 1}",
+        email: data["email"],
+        password: Devise.friendly_token[0,20]
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Space missing after comma.